pkg/privateendpoints/service.go

```diff
@@ -76,7 +77,7 @@ func (s *Service) ReconcileMcToWcApi(ctx context.Context) error {
 	//
 	for _, privateLink := range privateLinks {
 		logger.Info(fmt.Sprintf("Found private link %s", privateLink.Name))
-		manualApproval := !slice.Contains(privateLink.AutoApprovedSubscriptions, s.privateEndpointsScope.GetSubscriptionID())
+		manualApproval := !slice.Contains(util.ConvertToStringSlice(privateLink.AutoApprovedSubscriptions), s.privateEndpointsScope.GetSubscriptionID())
```

Comment:
I had to add a helper function converting privateLinks, because I had to update `github.com/giantswarm/cluster-api-provider-azure`

There's no native function included, that's why I had to come up with something

Reply:
private links are in `github.com/giantswarm/cluster-api-provider-azure v1.12.4` pointer string slices